### PR TITLE
[Upgrading] Recommend to delete cache when upgrading to a major version

### LINKED ISSUE #19437: Docs for Upgrading Cause Kernel Errors
I was referred here by the symfony/symfony project.  They're saying the upgrading docs are defective because it never instructs developers to clear both caches between major upgrades.

https://github.com/symfony/symfony/issues/53566#issuecomment-1898595734

https://symfony.com/doc/current/setup/upgrade_major.html

The result is dev machines may have a broken code environment after 2 or more major upgrades and it becomes impossible to use the `cache:clear` command.

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #19437.

### CODE REVIEW

File: setup/upgrade_major.rst [hunk 1]
```diff
@@ -182,6 +182,16 @@ Next, use Composer to download new versions of the libraries:
 
     $ composer update "symfony/*"
 
+It's also recommended to clear the cache after updating to a new major version:
+
+.. code-block:: terminal
+
+    # this deletes the development cache in your local machine
+    $ php bin/console cache:clear
```
Comment (xabbuh) [inline setup/upgrade_major.rst]: Running the `cache:clear` command will not solve the situation described in the linked issue if I understand it correctly. Being able to run this command means that your console application needs to be able to boot which can lead to the described issues if the dumped container is not compatible with your vendor code. The only solution to fix it is to remove the cache directory from the filesystem and to warm up the cache afterwards.

Comment (wouterj) [reply setup/upgrade_major.rst]: Yes, we should either recommend `rm -rf var/cache/*` or tell people to run `bin/console cache:clear --no-warmup` before the upgrade

Comment (javiereguiluz) [reply setup/upgrade_major.rst]: I've just reworded as you proposed. Thanks for the review!

**Review by alamirault — APPROVED** [review]

**Comment by miqrogroove** [comment]
Excellent, thank you.

**Pull request merged**